makeflow/src/parser_jx.c

```diff
@@ -93,36 +93,25 @@ static int resources_from_jx(struct hash_table *h, struct jx *j) {
 }
 
 static int file_from_jx(struct dag_node *n, int input, struct jx *j) {
-	char *path;
-	if (jx_match_string(j, &path)) {
-		if (input) {
-			debug(D_MAKEFLOW_PARSER, "Input %s", path);
-			dag_node_add_source_file(n, path, NULL);
-		} else {
-			debug(D_MAKEFLOW_PARSER, "Output %s", path);
-			dag_node_add_target_file(n, path, NULL);
-		}
-		free(path);
-		return 1;
-	} else if (jx_istype(j, JX_OBJECT)) {
-		const char *name = jx_lookup_string(j, "name");
-		const char *source = jx_lookup_string(j, "source");
-		if (!name) {
-			debug(D_MAKEFLOW_PARSER, "File lacks a name");
-			return 0;
-		}
-		if (input) {
-			debug(D_MAKEFLOW_PARSER, "Input %s, remote name %s", name, source ? source : "NULL");
-			dag_node_add_source_file(n, name, source);
-		} else {
-			debug(D_MAKEFLOW_PARSER, "Output %s, remote name %s", name, source ? source : "NULL");
-			dag_node_add_target_file(n, name, source);
-		}
-		return 1;
-	} else {
-		debug(D_MAKEFLOW_PARSER, "Input must be a string or object");
+	if (!jx_istype(j, JX_OBJECT)) {
+		debug(D_MAKEFLOW_PARSER|D_NOTICE, "File must be specified as an object");
+		return 0;
+	}
+
+	const char *path = jx_lookup_string(j, "path");
+	const char *remote = jx_lookup_string(j, "execution_path");
+	if (!path) {
+		debug(D_MAKEFLOW_PARSER|D_NOTICE, "File must have a path");
```

Comment:
I must have deleted this comment, should have been in the review.

Would it make sense to say "File must have key:value pair for 'path'" to clarify that this is JSON formatted values. This applies to the other comments in this file.